llvm/lib/SYCLLowerIR/LowerKernelProps.cpp

```diff
@@ -44,17 +44,31 @@ void processSetKernelPropertiesCall(CallInst &CI) {
 
   switch (PropID) {
   case property_ids::use_large_grf:
+  case property_ids::use_auto_grf:
     // TODO: Keep track of traversed functions to avoid repeating traversals
     // over same function.
-    llvm::sycl::utils::traverseCallgraphUp(F, [](Function *GraphNode) {
-      GraphNode->addFnAttr(::sycl::kernel_props::ATTR_LARGE_GRF);
-      // Add RegisterAllocMode metadata with arg 2 to the kernel to tell
-      // IGC to compile this kernel in large GRF mode. 2 means large.
+    llvm::sycl::utils::traverseCallgraphUp(F, [=](Function *GraphNode) {
+      auto AttrToAdd = PropID == property_ids::use_large_grf
```

Comment:
I wouldn't recommend to use `auto` here, because it is not immediately obvious which type it actually is - especially if you are looking at this file alone at has no idea where `ATTR_LARGE_GRF` comes from. ([LLVM Coding Standard](https://llvm.org/docs/CodingStandards.html#use-auto-type-deduction-to-make-code-more-readable)).

I feel like it is a `StringRef`, which makes me want to write a comment that string comparisons made down below are not that efficient as if you compared numeric IDs instead

Reply:
thx, i will probably delete this whole thing anyway but if not i will apply this